File: src/main/java/engineer/carrot/warren/thump/config/GeneralConfiguration.java
package engineer.carrot.warren.thump.config;

import com.google.common.collect.Lists;
import net.minecraftforge.common.config.Configuration;

public class GeneralConfiguration {
    private static final String CATEGORY = "general";

    public boolean logIrcToServerConsole = true;
    private static final String LOG_IRC_TO_SERVER_CONSOLE_KEY = "LogIRCToServerConsole";

    public GeneralConfiguration(Configuration configuration) {
        configuration.setCategoryPropertyOrder(CATEGORY, Lists.newArrayList(LOG_IRC_TO_SERVER_CONSOLE_KEY));
        this.logIrcToServerConsole = configuration.getBoolean(LOG_IRC_TO_SERVER_CONSOLE_KEY, CATEGORY, this.logIrcToServerConsole, "");
    }
}

File: src/main/kotlin/engineer/carrot/warren/thump/config/GeneralConfiguration.kt
package engineer.carrot.warren.thump.config

import net.minecraftforge.common.config.Configuration

class GeneralConfiguration(configuration: Configuration) {
    var obfuscateUserSourceFromMinecraft = true

    init {
        this.obfuscateUserSourceFromMinecraft = configuration.getBoolean(OBFUSCATE_USER_SOURCE_FROM_MINECRAFT, CATEGORY, this.obfuscateUserSourceFromMinecraft, OBFUSCATE_USER_SOURCE_FROM_MINECRAFT_COMMENT)
    }

    companion object {
        private val CATEGORY = "general"
        private val OBFUSCATE_USER_SOURCE_FROM_MINECRAFT = "ObfuscateUserSourceFromMinecraft"
        private val OBFUSCATE_USER_SOURCE_FROM_MINECRAFT_COMMENT = "Inserts a zero-width character in to source usernames going from Minecraft to IRC - commonly used to prevent unnecessary pings."
    }
}
